### ISSUE TITLE
Integrate deleteMany API

### ISSUE THREAD

**Issue by kathirsvn** [stargate/stargate-mongoose#28]
Add implementation for deleteMany API

Sample response from the server:

```
{
"status" { "deletedCount" : 100, "moreData" : true}
}
```

When `moreData` is received as `true` from the server, the driver should throw an error back to the client.

**Comment by vkarpov15** [comment]
Out of curiousity, under what circumstances would the server return `{ moreData: true }`?

**Comment by kathirsvn** [comment]
@vkarpov15 This logic needs to be discussed and improved further. I just created this ticket for tracking with the base idea we discussed for this API. 

Basically, the server will set a hard limit on number of docs that can be deleted in a deleteMany call. So, when a filter condition received finds more docs in the database than the limit, it will delete only the 'x' number of docs (i.e. the hard limit) and will return 'moreData' as true (i.e. more data is present on the database to delete for the given criteria). I'm not sure if it should be returned back as a soft notification to the client or an error since the threshold is reached.
